### PR TITLE
Don't reset before_context_ivars in reset_memoized

### PR DESCRIPTION
**Pull request by bootstraponline**
This isn't intended to be merged. Tracking down a potential memory issue related to #2095 I'm expecting this PR to fail on Travis.

### CODE REVIEW

**Comment by bootstraponline** [comment]
I expected to see the memory test fail, instead it's failing for some unknown reason.

> 1816 examples, 0 failures, 3 pending
>
> The command "script/run_build" exited with 2.
> https://travis-ci.org/rspec/rspec-core/jobs/87574215

**Comment by bootstraponline** [comment]
It looks like [master is failing with the same error.](https://travis-ci.org/rspec/rspec-core/builds/87571130). I guess the issue is flaky tests. This change should be safe to merge then.

**Comment by bootstraponline** [comment]
> lib/rspec/core/rake_task.rb
> RSpec::Support.define_optimized_require_for_rspec(:core) { |f| require_relative "../#{f}" }

shows up as missing in the coverage report which explains the exit code.

File: lib/rspec/core/rake_task.rb [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
 RSpec::Support.require_rspec_support "ruby_features"
 
+# :nocov:
 unless RSpec::Support.respond_to?(:require_rspec_core)
```
Comment (bootstraponline) [inline lib/rspec/core/rake_task.rb]: When I run locally on Ruby 2.23:

> ./script/rspec_with_simplecov spec --backtrace --format progress --profile --format progress

require_rspec_core is already defined so the unless block never executes (and counts against coverage).

**Comment by JonRowe** [comment]
:+1:

**Pull request merged**